src/framework/mlt_frame.h

```diff
@@ -97,6 +97,8 @@ struct mlt_frame_s
 	mlt_deque stack_audio;   /**< \private the audio processing stack of operations and data */
 	mlt_deque stack_service; /**< \private a general purpose data stack */
 	int is_processing;       /**< \private indicates if a frame is or was processed by the parallel consumer */
+	struct mlt_image_s image;        /**< \private the frame image */
```

Comment:
This is the heart of the entire change set. I added mlt_image to the public interface. I see here that I labeled it private. But I access it directly in many modules. Maybe there should be a get function instead of direct access.

I also added mlt_audio to the frame for illustration.

Reply:
We should make a private member of the mlt_frame struct like we have done elsewhere and put things into it including:
- stack_image
- stack_audio
- stack_service
- is_processing
- image
- audio